compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -1047,6 +1050,49 @@ private STToken processIdentifierOrKeyword() {
         }
     }
 
+    private int getNextNonWSOrNonCommentChar() {
+        int lookaheadCount = 0;
+        while (!reader.isEOF()) {
```

Comment:
since `peek()` does not affect `reader.isEOF()` we need to find other terminator condition. Maybe we can check for `Character.MAX_VALUE`

Reply:
Updated 24cd46b8d7ae464cde4c0e7429eb645023dd0ff9